### PR TITLE
[stable/prometheus-operator] Allow annotations on prometheus service

### PR DESCRIPTION
**Pull request by KierranM**
#### What this PR does / why we need it:
Allows setting custom annotations on the prometheus service by setting `prometheus.service.annotations`.
The reason I'm making this change is that I have a use case where it is necessary to have an AWS NLB for the service which we have previously created using annotations on the service while using the stable/prometheus chart.

#### Special notes for your reviewer:

I've only added annotations to the prometheus service to keep the change small, but I'm happy to add them to other services/resources if desired.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by vsliouniaev** [comment]
I think it would be useful to add this to the alertmanager configuration too, grafana already has this capability

**Comment by gianrubio** [comment]
@KierranM thanks for your contribution! Would you mind to add the same annotation to alertmanager?

**Review by vsliouniaev — CHANGES_REQUESTED** [review]
Could you please sync up the values with the CI file as well? Apart form that I think it's good to go

**Comment by rajatvig** [comment]
@KierranM I also ended up creating a similar one at https://github.com/helm/charts/pull/8966. I am ok closing that out. Could you also add options to specify `service.type` and `service.nodePort` as well?

**Comment by vsliouniaev** [comment]
@KierranM could you bump the chart version again please?

**Comment by rajatvig** [comment]
@KierranM are you still working on it? my changes are blocked waiting on this.

**Comment by KierranM** [comment]
Sorry been away for a few days. Bumping the version now

**Comment by rajatvig** [comment]
Looks good. Will close my PR once this is in as there is not much extra in mine. Thanks @KierranM

**Review by vsliouniaev — APPROVED** [review]

File: stable/prometheus-operator/README.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ The following tables lists the configurable parameters of the prometheus-operato
 | `prometheus.ingress.labels` | Prometheus Ingress additional labels | `{}` |
 | `prometheus.ingress.hosts` | Prometheus Ingress hostnames | `[]` |
 | `prometheus.ingress.tls` | Prometheus Ingress TLS configuration (YAML) | `[]` |
+| `prometheus.service.type` |  Prometheus Service typre | `ClusterIP` |
```
Comment (paskal) [inline stable/prometheus-operator/README.md]: typre -> type

File: stable/prometheus-operator/templates/alertmanager/service.yaml [hunk 2]
```diff
@@ -6,6 +6,10 @@ metadata:
   labels:
     app: {{ template "prometheus-operator.name" . }}-alertmanager
 {{ include "prometheus-operator.labels" . | indent 4 }}
+  {{- if .Values.alertmanager.service.annotations }}
```
Comment (paskal) [inline stable/prometheus-operator/templates/alertmanager/service.yaml]: is indentation correct?

File: stable/prometheus-operator/templates/prometheus/service.yaml [hunk 3]
```diff
@@ -6,12 +6,20 @@ metadata:
   labels:
     app: {{ template "prometheus-operator.name" . }}-prometheus
 {{ include "prometheus-operator.labels" . | indent 4 }}
+  {{- if .Values.prometheus.service.annotations }}
```
Comment (paskal) [inline stable/prometheus-operator/templates/prometheus/service.yaml]: is indentation correct?

**Review by paskal — CHANGES_REQUESTED** [review]
Could you please look into my comments while we still didn't got lgtm tag?

**Comment by gianrubio** [comment]
please fix CI

**Comment by gianrubio** [comment]
@KierranM CI is still broken

**Pull request merged**